app/src/main/java/org/mozilla/fenix/tabstray/browser/AbstractBrowserTabViewHolder.kt

```diff
@@ -130,7 +130,8 @@ abstract class AbstractBrowserTabViewHolder(
     private fun updateTitle(tab: TabSessionState) {
         val title = tab.content.title.ifEmpty {
             tab.content.url
-        }
+                .take(MAX_URI_LENGTH)
+        }.take(MAX_URI_LENGTH) // We can use the max URI length for titles as well.
```

Comment:
This LGTM. We also have an extension function, `TabSessionState.toDisplayTitle()`, and I'm wondering if we should instead refactor that to be this logic instead and use that function here (it's apparently only used in the `InactiveTabs` Composable so that long title bug might be present there, too).

Reply:
Nice, I can update this to use that extension that does exactly what we need.

I'll add the `.take(MAX_URI_LENGTH)` here, but it should be within that extension function and I wouldn't want to change that as part of this fix.